### PR TITLE
Migrate almost all CLI tests to use the command based CLI 

### PR DESCRIPTION
**Pull request by keyboardDrummer**
### Description
- Migrate almost all CLI tests to use the command based CLI
- Tests that use options that are only available in the old CLI have not been migrated

### How has this been tested?
- The tests themselves have changed

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/c++/tuple.dfy.expect [hunk 1]
```diff
@@ -1,3 +1,3 @@
 
-Dafny program verifier finished with 6 verified, 0 errors
+Dafny program verifier finished with 7 verified, 0 errors
```
Comment (atomb) [inline Source/IntegrationTests/TestFiles/LitTests/LitTest/c++/tuple.dfy.expect:2]: Do you know why this changed? I don't think it's likely a problem, but it's odd that just changing to the new CLI would do this.

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/dafny0/GhostGuards.dfy [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // NONUNIFORM: need to add support for tooltips to the new CLI (if that makes sense)
```
Comment (atomb) [inline Source/IntegrationTests/TestFiles/LitTests/LitTest/dafny0/GhostGuards.dfy:1]: Is this comment obsolete now?

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/c++/tuple.dfy.expect (hunk 1 above)
Comment (atomb) [reply Source/IntegrationTests/TestFiles/LitTests/LitTest/c++/tuple.dfy.expect:2]: Oh, it's due to the change in default for definite assignment, isn't it?

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/dafny0/CompilationErrors.dfy [hunk 3]
```diff
@@ -1,8 +1,8 @@
-// RUN: %exits-with 3 %dafny /allowAxioms:0 "%s" > "%t"
+// RUN: %exits-with 3 %build "%s" > "%t"
 // RUN: %diff "%s.expect" "%t"
 
-type MyType  // compile error: abstract type
```
Comment (atomb) [inline Source/IntegrationTests/TestFiles/LitTests/LitTest/dafny0/CompilationErrors.dfy:4]: I'm confused about this change. It should still be a compile error, shouldn't it?

**Review by atomb — APPROVED** [review]
I'm glad we can get most of the tests working with the new test commands, and that it clarifies which tests need relaxed definite assignment. I left a few questions, but none of them are blocking.

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/c++/tuple.dfy.expect (hunk 1 above)
Comment (keyboardDrummer) [reply Source/IntegrationTests/TestFiles/LitTests/LitTest/c++/tuple.dfy.expect:2]: Could be, but also the previous number was amount of implementations (a concept almost invisible to the user), and the new number is amount of assertion batches (slightly visible).

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/dafny0/CompilationErrors.dfy (hunk 3 above)
Comment (keyboardDrummer) [reply Source/IntegrationTests/TestFiles/LitTests/LitTest/dafny0/CompilationErrors.dfy:4]: I don't know. The type is not used by the rest of the code, maybe that matters.

File: Source/IntegrationTests/TestFiles/LitTests/LitTest/dafny0/GhostGuards.dfy (hunk 2 above)
Comment (keyboardDrummer) [reply Source/IntegrationTests/TestFiles/LitTests/LitTest/dafny0/GhostGuards.dfy:1]: The message after NONUNIFORM is obsolete yes.

**Review by atomb — DISMISSED** [review]

**Review by MikaelMayer — APPROVED** [review]
Approved last two commits

**Pull request merged**